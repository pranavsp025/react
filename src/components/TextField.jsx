import { forwardRef, useEffect, useState } from "react";

const TextField = forwardRef((props,ref) => {
    
    const [text,setText]=useState('')

    // useEffect(()=>{
    //     if(text.length>10){
    //         console.log("error")
    //         isError(true)
    //     }
    // },[text,isError])

    return(
        <span>
            <label htmlFor="uname" >{props.label}</label>
            <input ref={ref} type={props.type} value={props.value} onChange={props.onChange}  style={{borderColor:props.isError ? 'Red':'#03aeee'}} placeholder={props.placeholder}/>
        </span>
    )
})

export default TextField;