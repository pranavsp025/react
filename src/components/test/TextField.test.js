import React from "react"
import { render,fireEvent } from "@testing-library/react"
import TextField from "../TextField";

describe("Check if TextField works properly", () =>{
    test("Check if TextField rendered correctly",()=>{
        const {getByTestId} = render(<TextField />);

        const element = getByTestId('textField-test-id');


        expect(element).toBeTruthy()
    })
    
    test("Check if value displayed properly",()=>{
        const value="value";
        const onChange = jest.fn();

        
        const { getByDisplayValue } = render(<TextField value={value} onChange={onChange}/>)
        getByDisplayValue(value)

    })
    test("Check if label displayed properly",()=>{
        const label="label";
        
        const { getByLabelText } = render(<TextField label={label}/>)


    })
    test("Check if placeholder displayed properly",()=>{
        const placeholder = "placeholder";
        
        const { getByPlaceholderText } = render(<TextField placeholder={placeholder}/>)

        getByPlaceholderText(placeholder)

    })

    test("Check if onChange is triggerred", () => {
        const onChange = jest.fn();

        const {getByTestId} = render(<TextField onChange={onChange}/>);
        const element = getByTestId('textField-test-id');

        fireEvent.change(element);

        expect(onChange).toHaveBeenCalledTimes(0)
    })

    test("Check is snapshots is matched properly",() => {
        const onChange = jest.fn();
        const value="value";
        const placeholder = "placeholder";
        const type = "type";

        const { asFragment } = render(<TextField value={value} type={type} placeholder={placeholder} onChange={onChange}/>);

        expect(asFragment()).toMatchSnapshot();
    })
})