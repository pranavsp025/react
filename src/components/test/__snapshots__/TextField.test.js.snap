// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if TextField works properly Check is snapshots is matched properly 1`] = `
<DocumentFragment>
  <span>
    <label
      for="uname"
    />
    <input
      data-testid="textField-test-id"
      placeholder="placeholder"
      style="border-color: #03aeee;"
      type="type"
      value="value"
    />
  </span>
</DocumentFragment>
`;
